stm32-modules/include/common/core/gcode_parser.hpp

```diff
@@ -121,6 +122,201 @@ auto parse_value(const Input& start_from, Limit stop_at)
     return std::make_pair(std::optional<ValueType>(value), after);
 }
 
+// Gcode argument base required values
+template <typename Arg>
+concept GCodeArgument = requires(Arg& a) {
+    // Needs a flag for whether it's required
+    {std::is_same_v<decltype(Arg::required), bool>};
+    // Variable within the argument - flag for if it is present
+    {std::is_same_v<decltype(a.present), bool>};
+};
+
+// Gcode argument that has a prefix
+template <typename Arg>
+concept GCodeArgumentWithPrefix = requires(Arg& a) {
+    // Prefix needs to be iterable
+    {Arg::prefix.cbegin()};
```

Comment:
```suggestion
    {std::is_same_v<std::remove_cvref<decltype(*Arg::prefix.cbegin())>, char>}
```
?

Reply:
oh it also might be better to use `std::begin` and `std::end` rather than the member fns